sds_data_manager/lambda_code/SDSCode/queries.py

```diff
@@ -28,14 +33,62 @@ def lambda_handler(event, context):
     logger.info(f"Context: {context}")
 
     logger.info("Received event: " + json.dumps(event, indent=2))
-    # TODO: Replace result using RDS
-    search_result = ""
-    logger.info("Query Search Results: " + json.dumps(search_result))
+
+    # add session, pick model like in indexer and add query to filter_as
+    query_params = event["queryStringParameters"]
+
+    # select the file catalog for the query
+    query = select(models.FileCatalog.__table__)
+    # get a list of all valid search parameters
+    valid_parameters = [
+        column.key
+        for column in models.FileCatalog.__table__.columns
+        if column.key not in ["id", "status_tracking_id"]
+    ]
+    # go through each query parameter to set up sqlalchemy query conditions
+    for param, value in query_params.items():
+        # confirm that the search parameter is valid
+        if param not in valid_parameters:
+            response = {
+                "statusCode": 400,
+                "body": json.dumps(
+                    f"{param} is not a valid search parameter. "
+                    + f"Valid search parameters are: {valid_parameters}"
```

Comment:
Do we want to say "query" instead of "search" here? I was wondering if we wanted to change the endpoint to `/search` actually :) But, I guess we are "querying" and not "searching"?

Reply:
Yeah, I'll change that to query